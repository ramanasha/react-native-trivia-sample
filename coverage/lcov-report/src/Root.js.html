<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Root.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">src</a> Root.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/45</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/15</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >/** @flow */<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
import Reac<span class="cstat-no" title="statement not covered" >t from 'react';<span class="cstat-no" title="statement not covered" ></span></span>
import { St<span class="cstat-no" title="statement not covered" >oreProvider } from 's</span>rc/redux';
import { Root as<span class="cstat-no" title="statement not covered" > RootContainer, StylePr</span>ovider } from 'native-base';
import { Spla<span class="cstat-no" title="statement not covered" >sh } from 'src/screens'</span>;
import { Provide<span class="cstat-no" title="statement not covered" >r } from 'react-redux';</span>
import { Connect<span class="cstat-no" title="statement not covered" >edNavigator } from 'src/na</span>vigation';
import { The<span class="cstat-no" title="statement not covered" >me } from 'src/styles'</span>;
import { Per<span class="cstat-no" title="statement not covered" >sistGate } from 'redux-persist/src/integration/</span>react';
import config<span class="cstat-no" title="statement not covered" > from '../config/default.json';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span>
&nbsp;
<span class="cstat-no" title="statement not covered" >const [store, persistor] = StoreProvider.init</span>(config);<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span>
&nbsp;
const Roo<span class="cstat-no" title="statement not covered" >t = funct<span class="fstat-no" title="function not covered" >ion (</span>): React$Element&lt;Provider&gt; {</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
        &lt;Provider store={store}&gt;
            &lt;StyleProvider style={Theme.getTheme(Theme.variables)}&gt;
                &lt;PersistGate
                    loading={&lt;Splash /&gt;}
                    persistor={persistor}
                    on<span class="fstat-no" title="function not covered" >BeforeLift={(</span>) <span class="cstat-no" title="statement not covered" >=&gt; StoreProvider.serverSync(store)}</span>
                &gt;
                    &lt;RootContainer&gt;
                        &lt;ConnectedNavigator /&gt;
                    &lt;/RootContainer&gt;
                &lt;/PersistGate&gt;
            &lt;/StyleProvider&gt;
        &lt;/Provider&gt;
    );
};<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
export default Root;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
