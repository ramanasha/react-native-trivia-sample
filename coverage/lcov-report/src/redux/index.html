<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/redux
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>59/75</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">76.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>23/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>16/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.46% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>55/71</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="reducers.js"><a href="reducers.js.html">reducers.js</a></td>
	<td data-value="97.37" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.37" class="pct high">97.37%</td>
	<td data-value="38" class="abs high">37/38</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="97.3" class="pct high">97.3%</td>
	<td data-value="37" class="abs high">36/37</td>
	</tr>

<tr>
	<td class="file high" data-value="selectors.js"><a href="selectors.js.html">selectors.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="state.js"><a href="state.js.html">state.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file low" data-value="store.js"><a href="store.js.html">store.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
