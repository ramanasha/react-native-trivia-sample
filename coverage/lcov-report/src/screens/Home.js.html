<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/screens/Home.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/screens</a> Home.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import React from 'react';
import { Container, Header, Footer, Grid, Row, H1, Text, Button, Body, Icon } from 'native-base';
import { Styles } from 'src/styles';
import { Routes, navigateTo } from 'src/navigation';
import { connect } from 'react-redux';
import type { RootState } from 'src/redux/state';
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate: RootState) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
        questionCount: state.questions.items.length,
    };
};
&nbsp;
class HomeView extends React.Component&lt;{ questionCount: number, navigation: Object }&gt; {
    startQuiz = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
        const { navigation } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >        navigateTo(navigation, Routes.Quiz, { questionId: 0 });</span>
    };
&nbsp;
<span class="fstat-no" title="function not covered" >    re</span>nder(): React$Element&lt;Container&gt; {
        const { questionCount } = <span class="cstat-no" title="statement not covered" >this.props;</span>
&nbsp;
        const introMessage = <span class="cstat-no" title="statement not covered" >questionCount &gt; 0 ? `You will be presented with ${questionCount} True or False questions.`</span>
            : 'Oops!, there are no Questions at the moment';
&nbsp;
<span class="cstat-no" title="statement not covered" >        return (</span>
            &lt;Container&gt;
                &lt;Header padder&gt;
                    &lt;H1 center bold&gt;Welcome to the Trivia Challenge!&lt;/H1&gt;
                &lt;/Header&gt;
                &lt;Grid style={Styles.padder}&gt;
                    &lt;Row style={Styles.center}&gt;
                        &lt;Text center&gt;{introMessage}&lt;/Text&gt;
                    &lt;/Row&gt;
                    {questionCount &gt; 0 &amp;&amp;
                    &lt;Row style={Styles.center}&gt;
                        &lt;Text center&gt;Can you score 100%?&lt;/Text&gt;
                    &lt;/Row&gt;
                    }
                &lt;/Grid&gt;
                &lt;Footer&gt;
                    &lt;Body&gt;
                        &lt;Button
                            bordered
                            large
                            info
                            full
                            style={Styles.noPadding}
                            disabled={questionCount &lt; 1}
                            onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.startQuiz()}</span>
                        &gt;
                            &lt;Text uppercase&gt;Begin&lt;/Text&gt;
                            &lt;Icon active name="play-circle-outline" style={[Styles.icon, Styles.noMargin]} /&gt;
                        &lt;/Button&gt;
                    &lt;/Body&gt;
                &lt;/Footer&gt;
            &lt;/Container&gt;
        );
    }
}
&nbsp;
export default connect(mapStateToProps)(HomeView);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
