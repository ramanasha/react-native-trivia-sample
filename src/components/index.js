// @flow

import QuestionBox from 'src/components/QuestionBox';
import ResultItem from 'src/components/ResultItem';
import List from 'src/components/List';
import AutoHeightWebView from 'react-native-autoheight-webview';

export {
    QuestionBox,
    ResultItem,
    List,
    AutoHeightWebView,
};
