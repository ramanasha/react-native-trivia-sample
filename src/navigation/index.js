// @flow

import ConnectedNavigator, { Routes, getStateForAction } from 'src/navigation/ConnectedNavigator';
import { navigateTo, getNavigationAction } from 'src/navigation/utils';
import type { RoutesType } from 'src/navigation/ConnectedNavigator';

export {
    ConnectedNavigator,
    getStateForAction,
    Routes,
    navigateTo,
    getNavigationAction,
};

export type {
    RoutesType,
};
