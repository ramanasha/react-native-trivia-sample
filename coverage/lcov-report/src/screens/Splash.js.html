<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/screens/Splash.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/screens</a> Splash.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/2</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import React from 'react';
import { StatusBar, Image } from 'react-native';
import { Container, Grid, Row, Col, Text } from 'native-base';
import { Styles, Images } from 'src/styles';
&nbsp;
const Splash = <span class="fstat-no" title="function not covered" >fu</span>nction (): React$Element&lt;Container&gt; {
<span class="cstat-no" title="statement not covered" >    return (</span>
        &lt;Container&gt;
            &lt;StatusBar translucent style={Styles.transparentBackground} barStyle="dark-content" /&gt;
            &lt;Grid&gt;
                &lt;Row style={Styles.center}&gt;
                    &lt;Col size={1} /&gt;
                    &lt;Col size={12}&gt;
                        &lt;Image source={Images.logo} style={[Styles.logo, Styles.center]} resizeMode="contain" /&gt;
                        &lt;Text center bigNote style={Styles.marginTop}&gt;Loading Questions...&lt;/Text&gt;
                    &lt;/Col&gt;
                    &lt;Col size={1} /&gt;
                &lt;/Row&gt;
            &lt;/Grid&gt;
        &lt;/Container&gt;
    );
};
&nbsp;
export default Splash;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
