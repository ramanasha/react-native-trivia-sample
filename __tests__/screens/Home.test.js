// @flow

import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../../src/screens';
import { Store } from '../setup';

describe('renders the Home screen correctly', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = shallow(<Home store={Store} />);
    });

    it('renders the Home container', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the Home elements', () => {
        expect(wrapper.dive()).toMatchSnapshot();
    });
});

