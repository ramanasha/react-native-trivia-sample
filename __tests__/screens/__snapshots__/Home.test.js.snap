// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Home screen correctly renders the Home container 1`] = `
<HomeView
  dispatch={[Function]}
  questionCount={2}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`renders the Home screen correctly renders the Home elements 1`] = `
<Styled(Container)>
  <Styled(Header)
    padder={true}
  >
    <Styled(H1)
      bold={true}
      center={true}
    >
      Welcome to the Trivia Challenge!
    </Styled(H1)>
  </Styled(Header)>
  <GridNB
    style={
      Object {
        "padding": 30,
      }
    }
  >
    <RowNB
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        center={true}
      >
        You will be presented with 2 True or False questions.
      </Styled(Text)>
    </RowNB>
    <RowNB
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)
        center={true}
      >
        Can you score 100%?
      </Styled(Text)>
    </RowNB>
  </GridNB>
  <Styled(Footer)>
    <Styled(Body)>
      <Styled(Button)
        bordered={true}
        disabled={false}
        full={true}
        info={true}
        large={true}
        onPress={[Function]}
        style={
          Object {
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <Styled(Text)
          uppercase={true}
        >
          Begin
        </Styled(Text)>
        <Styled(Icon)
          active={true}
          name="play-circle-outline"
          style={
            Array [
              Object {
                "fontSize": 30,
              },
              Object {
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
              },
            ]
          }
        />
      </Styled(Button)>
    </Styled(Body)>
  </Styled(Footer)>
</Styled(Container)>
`;
